src/actions.js

```diff
@@ -75,76 +74,72 @@ export function addCheckResult(
 // ASYNC (THUNK) ACTIONS.
 
 // Fetching the statuses.
-export function requestStatus(version: ?string): ThunkAction<void> {
+export function requestStatus(version: ?string) {
   const notifyChanges = (checkTitle, status) => {
     if (Notification.permission === 'granted') {
       new Notification(`${checkTitle} : status changed (${status}).`);
     }
   };
 
-  return function(dispatch: Dispatch, getState: GetState) {
+  return async function(dispatch: Dispatch, getState: GetState) {
     const versionToCheck = version || getState().version;
     if (!versionToCheck) {
       return;
     }
     // Save previous results so we can check if something changed.
     const prevResults = getState().checkResults;
     dispatch(setVersion(versionToCheck));
-    getReleaseInfo(versionToCheck)
-      .then(releaseInfo => {
-        dispatch(updateReleaseInfo(releaseInfo));
-        releaseInfo.checks.map((check: CheckInfo) => {
-          return checkStatus(check.url).then(result => {
-            // Detect if the result changed, and notify!
-            const prevResult = prevResults[check.title];
-            if (prevResult && prevResult.status !== result.status) {
-              notifyChanges(check.title, result.status);
-            }
-            dispatch(addCheckResult(check.title, result));
-          });
-        });
-      })
-      .catch((err: string) => {
-        console.error(
-          `Failed getting the release info for ${versionToCheck}`,
-          err,
-        );
-      });
+    const releaseInfo = await getReleaseInfo(versionToCheck);
+    dispatch(updateReleaseInfo(releaseInfo));
+    const checks = releaseInfo.checks.map(async (check: CheckInfo) => {
```

Comment:
Here you can do some destructuring. For example have `({url, title}: CheckInfo)` in the signature, and avoid having `check.title`, `check.url`. 

The way it is it's perfectly fine. with niko we used it a lot :) it's a matter of taste,